### ISSUE TITLE
"metadata" is not supported in app/.

### ISSUE THREAD

**Issue by alextseitlin** [vercel/next.js#46404]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:54 PST 2021; root:xnu-8019.61.5~1/RELEASE_X86_64
    Binaries:
      Node: 18.14.2
      npm: 9.5.0
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.2.2-canary.0
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### Link to the code that reproduces this issue

https://stackblitz.com/edit/vercel-next-js-ng1njf?file=next.config.js

### To Reproduce

Added Metadata to page as it says in the docs https://beta.nextjs.org/docs/api-reference/metadata

"Metadata can be used in any page.js or layout.js file. This API reference includes every possible option for the returned metadata object."

### Describe the Bug

Metadata works only in layout.js file and in page.js gives an error

### Expected Behavior

work

**Comment by erwanvivien** [comment]
Not a bug: 
https://stackblitz.com/edit/vercel-next-js-ng1njf?file=app%2Fpage.jsx%3AL1

Please see #46372

**Comment by alextseitlin** [comment]
Thanks a lot!!. But I would love to see more clear errors because the error contradicts what the docs says and it is very confusing =)

**Comment by erwanvivien** [comment]
I agree!

**Comment by alextseitlin** [comment]
> Not a bug: https://stackblitz.com/edit/vercel-next-js-ng1njf?file=app%2Fpage.jsx%3AL1
> 
> Please see #46372

after removing 'use client' and making it Server Component I now get another issue https://github.com/vercel/next.js/issues/46407 =(

Dont understand what is the write what to add meta then... always have a layout file? because I plan to use useState a lot....

**Comment by DuCanhGH** [comment]
@alextseitlin have you added `<head></head>` to your root layout?

**Comment by alextseitlin** [comment]
> 

Amazing! this fixes the Hydration issue. But now I just dont understand the new 13v workflow... I cant use meta in all page.js because some will have useState. If I will add meta in page.js and sometimes in layout.js it will be not consistent. So should I then just create an extra layout file for each page just for the meta? Or is there a better way?

**Comment by alextseitlin** [comment]
> @alextseitlin have you added `<head></head>` to your root layout? Like so:
> 
> ```js
> export default function RootLayout({ children }) {
>   return (
>     <html lang="en">
>       <head></head>
>       <body>{children}</body>
>     </html>
>   );
> }
> ```

Amazing! this fixes the Hydration issue. But now I just dont understand the new 13v workflow... I cant use meta in all page.js because some will have useState. If I will add meta in page.js and sometimes in layout.js it will be not consistent. So should I then just create an extra layout file for each page just for the meta? Or is there a better way?

**Comment by DuCanhGH** [comment]
@alextseitlin you shouldn't turn `page.js` and `layout.js` files into client components. Instead, you should create separate client components to isolate parts of your page that use `useState`, `useEffect`,...
For example, in your `page.js`:
```
import { Button } from "./Button.js";

export const metadata = {
    // stuff...
}

export default function Page() {
    return (
        <>
            <p>Hello world!</p>
            <Button />
        </>
    );
}
```
and your `Button.js` next to it:
```
"use client";
import { useState } from "react";

export const Button = () => {
    const [state, setState] = useState(0);
    return <button onClick={() => setState((_o) => _o + 1)}>{state}</button>;
};
```

**Comment by huozhi** [comment]
Hi, it's caused by the metadata exports being defined in client components, it's only allow to use in server components for now. The incorrect error message is been fixed in #46334

**Comment by erwanvivien** [comment]
@huozhi This is the same issue that you closed (#46372)

**Comment by erwanvivien** [comment]
Lmao sorry for the ping :)
